application/app/Http/Requests/UpdateInstitutionUserRequest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Enums\PrivilegeKey;
+use App\Models\Department;
+use App\Models\InstitutionUser;
+use App\Models\Role;
+use App\Rules\ModelBelongsToInstitutionRule;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Auth;
+
+/**
+ * @property string forename
+ * @property string surname
+ * @property string email
+ * @property string phone
+ * @property array<string> roles
+ * @property string department
+ * @property string vendor
+ */
+class UpdateInstitutionUserRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
+        abort_unless(InstitutionUser::whereId($this->getInstitutionUserId())->exists(), 404);
+
+        return filled($targetInstitutionId = $this->findInstitutionUserInstitutionId())
+            && filled($tokenInstitutionId = Auth::user()?->institutionId)
+            && Auth::hasPrivilege(PrivilegeKey::EditUser->value)
+            && $tokenInstitutionId === $targetInstitutionId;
+    }
+
+    public function rules(): array
+    {
+        return [
+            'user' => ['array', 'min:1'],
+            'user.forename' => 'filled',
+            'user.surname' => 'filled',
+            'email' => 'email',
+            'phone' => 'regex:/\+?[\s\d]+\d/',
```

Comment:
@DanBondarenko it makes sense to use more strict validation of the phone number, as for me. Based on this regex the phone number `+380` is a valid one.

Reply:
@kadmit Hi, thank you for the suggestion. 

I personally wouldn’t want to maintain a list of country calling codes in our codebase. It’s subject to changes (e.g. [Kazakhstan is changing its code in 2025](https://www.rferl.org/a/kazakhstan-international-dialing-code/31471644.html)) and not worth the headache IMO.

If you consider this a **[blocker]**, I’ll implement it – it’s your choice 🙂. 
But just for the record: I don’t think it would be an efficient use of time.